File: 0035-search-insert-position/0035-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        n = len(nums)
        ans = n
        low = 0
        high = n-1

        while(low<=high) :
            mid = low + (high - low) // 2

            if(nums[mid] >= target) :
                ans = mid
                high = mid-1
            else : low = mid+1
        return ans


        

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int n = nums.length;
        int low = 0;
        int high = n-1;
        int index = n;

        while(low<=high) {
            int mid = low + (high - low) / 2;
            if(nums[mid] >= target) {
                index = mid;
                high = mid -1;
            }
            else low = mid + 1;
        }
        return index;
    }
}